### DISCUSSION TITLE
M3U Playlists on SMB shares being migrated in new MA version?

### DISCUSSION THREAD

**Discussion by corywindsor** [music-assistant/support#2226] [Q&A] [answered]
Can someone confirm that with the newest versions of MA (2.0.0rc1 and maybe a couple before that) M3U playlists that reside on SMB shares seem to be automatically migrated to music assistant playlists? 

Normally I had M3U files to play specific tracks (using filesystem_smb://track/ as the format), I generate these using a playlist generator that can look at the library a grab x songs from each artist, etc. I would have these on my SMB share, and music assistant would load them and I could play them normally. 

As of this latest release I see it trying to do something to these files, I typically have music assistant only read-only access to my music library, so it was failing out trying to complete this. I gave it read and write access for testing, and it does seem to be migrating playlists from the SMB share to music assistant. It also seems to get rid of them from the SMB share as well.

The log of the event is below, it did this to any M3U playlist I gave it write access to. Sorry if I missed the discussion on this, I didn't see any reference to it. 

2024-05-09 15:26:30.743 INFO (MainThread) [music_assistant.providers.filesystem_smb] Migrated playlist xxplaylistnamexx

**Comment by OzGav** [comment, marked as answer]
Yes that is the case. All playlists are now (actually from quite a few betas ago) stored in the MA DB.

**Reply by corywindsor** [reply]
It seemed like this had a lot to do with having write access to the share, which is referenced in the documentation. Once I've done that, the M3U's are able to be stored on the filesystem. Removing the filesystem_smb://track/ from the beginning of each entry in the playlist, leaving only the end of the line. e.g.

Original:
filesystem_smb://track/Oasis/1994 - Definitely Maybe/Oasis - 06. Supersonic.mp3

New: 
Oasis/1994 - Definitely Maybe/Oasis - 06. Supersonic.mp3

Seems to work now, the playlists are scanned and play through the system.
